desc/optimize/least_squares.py

```diff
@@ -348,7 +353,7 @@ def lsqtr(  # noqa: C901
             success, message = check_termination(
```

Comment:
@f0uriest can you also solve this TODO? I am pretty sure that this has to be taken outside. Because the while loop condition already checks the only important criteria for the trust region problem. Stopping the entire optimization based on a step that is not even the step we are taking might be causing unintended termination?

Reply:
See response on #1395, we do need this check here to avoid infinite loops.

Reply:
Can you delete it then?